From 0966bfaef7c8171c1214ffbbe9de8782e602837d Mon Sep 17 00:00:00 2001
From: Yuval Caduri <cyuval@marvell.com>
Date: Tue, 1 May 2018 17:13:13 +0300
Subject: [PATCH 089/386] dts: marvell: musdk: add cma memory node

Continuous memory allocation module used to allocate memory for
MUSDK.

Change-Id: I19cc0377d787b997dbb44d90d1474670eeaa5c98
Signed-off-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5863
Reviewed-by: Nadav Haklai <Nadav.Haklai@cavium.com>
Reviewed-by: Liron Himi <liron.himi@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: sa_ip-sw-jenkins
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../bindings/net/marvell-musdk-cma.txt           | 16 ++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi     |  4 ++++
 arch/arm64/boot/dts/marvell/armada-ap80x.dtsi    |  5 +++++
 3 files changed, 25 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/marvell-musdk-cma.txt

diff --git a/Documentation/devicetree/bindings/net/marvell-musdk-cma.txt b/Documentation/devicetree/bindings/net/marvell-musdk-cma.txt
new file mode 100644
index 000000000000..4aa66bbdce5c
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/marvell-musdk-cma.txt
@@ -0,0 +1,16 @@
+* Marvell MUSDK continuous memory allocation driver (BM)
+
+Required properties:
+
+- compatible: should be "marvell,musdk-cma".
+
+Optional properties:
+
+- dma-coherent
+
+Example:
+
+musdk_cma {
+	compatible = "marvell,musdk-cma";
+	dma-coherent;
+};
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 174873cd919b..20196fde4c38 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -443,6 +443,10 @@
 				      <0x1da0000 0x20000>; /* GICV */
 				interrupts = <GIC_PPI 9 IRQ_TYPE_LEVEL_HIGH>;
 			};
+
+			musdk_cma {
+				compatible = "marvell,musdk-cma";
+			};
 		};
 
 		pcie0: pcie@d0070000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
index d05c5df871ee..bd310f04464d 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap80x.dtsi
@@ -439,6 +439,11 @@
 				reg = <0x610fcc 0x4>;
 			};
 
+			musdk_cma {
+				compatible = "marvell,musdk-cma";
+				dma-coherent;
+			};
+
 			agnic-plat {
 				compatible = "marvell,armada-giu-nic";
 				reg = <0x6F00A0 0x8>;
-- 
2.17.1

